enhancements/insights/prometheus-alerts-for-insights-recommendations.md

```diff
@@ -0,0 +1,277 @@
+   extension of the current behavior, where we only look at the number of
+   active recommendations per severity level.)
+3. Prometheus metrics indicating the status of recommendations being active are
+   set accordingly. Only metrics for active recommendations are reported during
+   the metric collection. If a recommendation is no longer active, the metric
+   will not be collected, and the in-cluster Prometheus is expected to delete
+   it, which in turns should cancel the corresponding alert.
+4. The metric will have labels containing the ID, severity (based on the total
+   risk property), a human-readable name, and a link to a detailed description
+   of the recommendation and its remediation steps (Insights Advisor URL
+   generated based on the cluster ID and rule ID by IO) and its value will be
+   0/1 to indicate if the recommendation is inactive/active respectively.
+5. There will be a generic/templated info-level alert present on the cluster,
+   which will present the user with brief information about the hitting
+   recommendation (its human-readable name and severity),
+   and the link for details and remediation steps.
+
+It is worth pointing out that if any key part of the Prometheus stack is not
+running as expected on the cluster, then this feature will not work. Prometheus
+manages both the metrics and the alerts, so without it, no alerts can be
+generated by IO.
+
+An example of what the Insights recommendation metric may look like:
+```text
+insights_recommendation_active{rule_id="ccx_rules_ocp.external.rules.empty_prometheus_db_volume.report",error_key="PROMETHEUS_DB_VOLUME_IS_EMPTY",severity="moderate",description="Prometheus metrics data will be lost when the Prometheus pod is restarted or recreated",info_url="http://example.com/"} 1646416494
+```
+
+**Unhealthy Insights Operator scenario:**
+
+- If something goes wrong with IO, we would not want the Insights alerts to
+  remain hanging on the cluster indefinitely.
+- The alert should only fire if IO has been healthy at any point in the past 10
+  minutes. Healthy, in this context, means that there is no major error
+  happening (crashlooping, degraded operator status, etc.). A temporary error
+  state should not cause the alerts to be disabled.
+- If the user intentionally disables IO, the alerts should never fire. If IO is
+  disabled while some of the Insights alerts are firing, the alerts should
+  immediately stop firing (the in-cluster will detect that IO is not active
+  during its period scraping, which will remove the Insights metrics from its
+  database).
+- If IO was disabled by an accident, the user will be notified via the existing
+  `InsightsDisabled` alert, which is defined as a part of the Insights Operator.
+
+**Unhealthy Smart Proxy scenario:**
+
+If the processing of a successfully uploaded archive takes unexpectedly long, a
+Prometheus metric indicating this delay should be set. This would most likely be
```

Comment:
> If the processing of a successfully uploaded archive takes unexpectedly long, a Prometheus metric indicating this delay should be set

I would reuse our existing `InsightsDownloadDegraded` cluster operator condition in this case and then introduce a new alert based on this condition as discussed. So I wouldn't create any new metric.

Reply:
Sure, that makes sense, and it will be easier to manage together instead of dealing with two separate indicators of the download delay.